### PR TITLE
PE-D Bug Fixing

### PR DESCRIPTION
**Pull request by kanna-1**
In this PR, the following PED-D issues are resolved and how:

1. #193 (By updating UG to include a tip to undo a clear command)
2. #191 (By updating UG to warn users that the command history is reset upon exiting HealthSync)
3. #190 (By updating UG to warn users that the identifications are not editable)
4. #162 (By adding the addPersonToIndex and updating DeleteCommand.java)
5. #163 (By updating UG)
6. #160 (By updating UG to warn users that duplicate names and ID are not allowed)
7. #218 (By adding new checks)

### CODE REVIEW

File: src/test/java/seedu/address/logic/commands/EditCommandTest.java [hunk 1]
```diff
@@ -167,11 +167,23 @@ public void execute_noChangeInEditFields_throwsCommandException() {
...
+    }
+
     @Test
     public void undo_successfulEditCommand() throws CommandException {
         Model model = new ModelManager();
-        Person originalPerson = new PersonBuilder().build();
-        Person editedPerson = new PersonBuilder().withName(VALID_NAME_AMY).withPhone(VALID_PHONE_BOB).build();
+        Person originalPerson = new PersonBuilder().withMedical(VALID_MEDICALHISTORY).build();
+        Person editedPerson = new PersonBuilder().withName(VALID_NAME_AMY).withPhone(VALID_PHONE_BOB)
+                .withMedical(VALID_MEDICALHISTORY).build();
```
Comment (nubnubyas) [inline src/test/java/seedu/address/logic/commands/EditCommandTest.java:186]: May I check why this change was made? Asking this because in `EditCommand.java`, I do not see a code change related.

**Review by nubnubyas — COMMENTED** [review]
Good implementation of `undo` using `index` to restore original ordering.  Only have one comment regarding whether a change is required. Check my comment.

**Review by hyc17003 — COMMENTED** [review]
Looks good! Good call standardising 'ID', think later on we may need to change all instances of IC, IC number etc to ID too. LGTM!

File: src/test/java/seedu/address/logic/commands/EditCommandTest.java (hunk 1 above)
Comment (kanna-1) [reply src/test/java/seedu/address/logic/commands/EditCommandTest.java:186]: Thank you for this review. It is a consequence of a bug I did not realize existed after reading this comment. I will be updating the PR after fixing the bug.

**Comment by kanna-1** [comment]
Merging.
